modules/assim.batch/R/autoburnin.R

```diff
@@ -0,0 +1,51 @@
+#' @name getBurnin
+#' @title Calculate burnin value
+#' @description Use `coda::gelman_plot` to calculate first iteration number where Gelman diagnostic never exceeds the specified threshold.
+#' @param jags_out List of MCMC sample matrices or `mcmc.list` object
+#' @param threshold Maximum value of Gelman diagnostic
+#' @param ... Other parameters to `gelman.plot`
+#' @examples
+#'      library(coda)
+#'      data(line)
+#'      burnin <- getBurnin(line, threshold = 1.05)
+#' @author Michael Dietze, Alexey Shiklomanov
+#' @export
+getBurnin <- function(jags_out, threshold = 1.1, ...) {
+    library(coda)
+    if (length(find("logger.info")) == 0) {
+        msg <- message
+    } else {
+        msg <- logger.info
+    }
+    if (!is.mcmc.list(jags_out)) jags_out <- makeMCMCList(jags_out)
+    png("/dev/null")
+    GBR <- try(gelman.plot(jags_out, autoburnin=FALSE, ...))
+    dev.off()
+    if (class(GBR) == "try-error") {
+        msg("Unable to calculate Gelman diagnostic. Assuming no convergence.")
+        return(1)
+    }
+    burnin <- GBR$last.iter[tail(which(rowSums(GBR$shrink[,,2] > threshold) > 0), 1) + 1]
+    if (is.na(burnin) | length(burnin) == 0) {
```

Comment:
@ashiklom do you ever get `length(burnin) == 0`? if yes `if (is.na(burnin) | length(burnin) == 0) ` would throw an error because `is.na(burnin)` would return `logical(0)` that `if` doesn't like